### PR TITLE
OLM go based CSV manifest generation

### PR DESCRIPTION
**Pull request by nitishkumar4**
### Steps to generate ClusterServiceVersion.yaml

Edit manifest file variables:
`VERSION ?= 2.5.0`
Note: Version has to of format `integer.integer.integer`
`IMAGE_TAG_BASE ?= quay.io/ibm-spectrum-scale/ibm-spectrum-scale-csi-operator`

Update base operator deployment manifest (config/manager/manager.yaml)
`image: IMAGE_TAG_BASE:v0.0.0`

Update base CSV manifest (config/manifests/bases/ibm-spectrum-scale-csi-operator.clusterserviceversion.yaml)
`containerImage: IMAGE_TAG_BASE`
`replaces: ibm-spectrum-scale-csi-operator.v<LAST VERSION>`

Generate CRD, ClusterRoles and Clusterrolebindings.
`make manifests` 

Generate CSV file.
`make bundle`

Copy CSV to config/olm-catalog/<VERSION>
`make generate`

CSV and CRD will be generated and updated under `/config/olm-catalog/<VERSION>` directory.

### CODE REVIEW

**Comment by nitishkumar4** [comment]
Creating follow up issue to handle CR required fields during upgrade, which fails the upgrade via OLM, as OLM's own CRD validation mechanism compares the CR and CRD and doesn't apply the CRD if required values are present in the CR. Since status is a required value and user cannot modify it this creates issue while upgrading. This is already handled for CLI upgrade path.

**Review by madhuthorat — COMMENTED** [review]
Some of the changes here are already handled in https://github.com/madhuthorat/ibm-spectrum-scale-csi/tree/goop_code branch. That branch is ahead of go_operator branch.

Can you please raise another PR with similar changes for that branch ?

**Comment by nitishkumar4** [comment]
> Creating follow up issue to handle CR required fields during upgrade, which fails the upgrade via OLM, as OLM's own CRD validation mechanism compares the CR and CRD and doesn't apply the CRD if required values are present in the CR. Since status is a required value and user cannot modify it this creates issue while upgrading. This is already handled for CLI upgrade path.

This is fixed.

File: config/olm-catalog/ibm-spectrum-scale-csi-operator/2.4.0/ibm-spectrum-scale-csi-operator.v2.4.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -3,12 +3,12 @@ apiVersion: operators.coreos.com/v1alpha1
...
     capabilities: Basic Install
     categories: Storage
     certified: 'false'
-    containerImage: quay.io/ibm-spectrum-scale/ibm-spectrum-scale-csi-operator:v2.4.0
+    containerImage: quay.io/ibm-spectrum-scale/ibm-spectrum-scale-csi-operator:v2.5.0
```
Comment (madhuthorat) [inline config/olm-catalog/ibm-spectrum-scale-csi-operator/2.4.0/ibm-spectrum-scale-csi-operator.v2.4.0.clusterserviceversion.yaml]: We don't need to update 2.4.0 CSV

**Review by madhuthorat — COMMENTED** [review]
Do we need bundle/ directory and its files ? Can they be avoided as we already have a copy at olm-catalog/

File: config/olm-catalog/ibm-spectrum-scale-csi-operator/2.4.0/ibm-spectrum-scale-csi-operator.v2.4.0.clusterserviceversion.yaml (hunk 1 above)
Comment (nitishkumar4) [reply config/olm-catalog/ibm-spectrum-scale-csi-operator/2.4.0/ibm-spectrum-scale-csi-operator.v2.4.0.clusterserviceversion.yaml]: yeah, removed it. The file got update during unit testing and I pushed the changes. Thanks.

**Comment by nitishkumar4** [comment]
> Do we need bundle/ directory and its files ? Can they be avoided as we already have a copy at olm-catalog/

we will need the directory to build the bundle directly using make commands instead of ansible script passed by testing team.

**Review by madhuthorat — APPROVED** [review]

**Pull request merged**